nix/overlays/checked-shell-script/checked-shell-script.nix

```diff
@@ -32,7 +40,7 @@ let
             exit 1
           fi
         ''
-        + text;
+        + "(${text})";
```

Comment:
This should run the remaining script in a sub-shell. The sub-shell will inherit almost everything - except the trap.